src/lib/components/wizard-navigation-bar.component.vertical.less

```diff
@@ -31,368 +31,179 @@
-    content: '';
-    position: absolute;
-    left: -(@dot-baseline-distance + @dot-height / 2);
-    top: @dot-height;
-    height: calc(100% ~'-' @dot-height);
-    width: 1px;
-  }
+.state-circle-hover(@dot-width, @dot-height, @dot-border-width) { }
+.state-circle-nohover(@dot-width, @dot-height, @dot-border-width) { }
 
-  .state-circle(@dot-width, @dot-height, @dot-border-width) {
-    position: absolute;
-    top: 0;
-    left: -(@dot-baseline-distance + @dot-width);
-    width: @dot-width;
-    height: @dot-height;
-    text-align: center;
-    vertical-align: middle;
-    line-height: @dot-height - 2 * @dot-border-width;
-    transition: 0.25s;
-    border-radius: 100%;
-  }
+.state-circle-with-border(@color) {
+  border-width: @border;
+  border-style: solid;
+  border-color: @color;
+}
+.state-circle-with-border-hover(@color) {
+  border-color: darken(@color, 10%);
+}
+.state-circle-with-border-nohover(@color) {
+  border-color: @color;
+}
 
-  .state-circle-hover(@dot-width, @dot-height, @dot-border-width) { }
-  .state-circle-nohover(@dot-width, @dot-height, @dot-border-width) { }
+.state-circle-with-border-and-content(@color) {
+  .state-circle-with-border(@arguments);
+  color: @color;
+}
+.state-circle-with-border-and-content-hover(@color) {
+  .state-circle-with-border-hover(@arguments);
+  color: darken(@color, 10%);
+}
+.state-circle-with-border-and-content-nohover(@color) {
+  .state-circle-with-border-nohover(@arguments);
+  color: @color;
+}
 
-  .state-circle-with-border(@dot-border-width, @circle-color) {
-    border-width: @dot-border-width;
-    border-style: solid;
-    border-color: @circle-color;
+.state-circle-with-background(@color) {
+  background-color: @color;
+}
+.state-circle-with-background-hover(@color) {
+  background-color: darken(@color, 5%);
+}
+.state-circle-with-background-nohover(@color) {
+  background-color: @color;
+}
+
+.state-circle-with-background-and-content(@color) {
+  .state-circle-with-background(@color);
+  color: black;
+}
+.state-circle-with-background-and-content-hover(@color) {
+  .state-circle-with-background-hover(@color);
+}
+.state-circle-with-background-and-content-nohover(@color) {
+  .state-circle-with-background-nohover(@color);
+}
+
+
+.define-state(@width: @big-dot-height, @height: @big-dot-height, @border: @dot-border-width) {
+  padding: (@distance-between-steps / 2) (@distance-between-steps / 2) (@distance-between-steps / 2) (@distance-between-steps / 2 + @dot-baseline-distance + @height);
+
+  li {
+    &:not(:last-child):before {
+      .line(@width, @height, @wz-color-default);
+    }
+
+    a {
+      min-height: @height;
+    }
+
+    .step-indicator {
+      .state-circle(@width, @height, @border);
+    }
+
+    a:hover .step-indicator {
+      .state-circle-hover(@width, @height, @border);
+    }
   }
-  .state-circle-with-border-hover(@dot-border-width, @circle-color) {
-    border-color: darken(@circle-color, 10%);
+
+  li .step-indicator {
```

Comment:
I'm in favor of adding this and the following blocks to the previous `li` block:
```less
li {
  .step-indicator {
    ...
  }

  &.current .step-indicator {
    ...
  }

  ...
}
```
I think this makes the code easier to read and understand.

Reply:
6370d56